src/utils/form-actions.ts

```diff
@@ -78,6 +78,26 @@ export async function saveTemplate(template: Omit<Template, 'id'>) {
   return data;
 }
 
+export async function getTemplate(
+  templateId: string
+): Promise<Template | undefined> {
+  const { data, errors } =
+    await getAmplifyBackendClient().models.TemplateStorage.get({
+      id: templateId,
+    });
+
+  if (errors) {
+    logger.error('Failed to get template', errors);
+  }
+
+  if (!data) {
+    logger.warn(`Failed to retrieve template for ID ${templateId}`);
+    return undefined;
+  }
+
+  return data as Template;
```

Comment:
does this work as `satisfies`?

Reply:
Ah probably not because `Template` is such a complex type. Ignore this.

Reply:
satisfies caused problems. i've added a zod call instead so we don't need the type cast